File: analyze_limp.py

import json
import math

# Load data from data.js
with open('data.js', 'r') as f:
    content = f.read()
    json_str = content.replace('const fallData = ', '').rstrip(';')
    data = json.loads(json_str)

def analyze_activity(subject, activity_id):
    samples = data[subject][activity_id]["Waist"]
    accs = [math.sqrt(s['acc_x']**2 + s['acc_y']**2 + s['acc_z']**2) for s in samples]
    
    peaks = []
    for i in range(1, len(accs) - 1):
        if accs[i] > accs[i-1] and accs[i] > accs[i+1] and accs[i] > 11.0: # Filter low-level noise
            peaks.append(accs[i])
            
    avg_peak = sum(peaks) / len(peaks) if peaks else 0
    # Calculate std dev manually
    std_peak = math.sqrt(sum((p - avg_peak)**2 for p in peaks) / len(peaks)) if peaks else 0
    
    return {
        "avg_peak": round(avg_peak, 2),
        "std_peak": round(std_peak, 2),
        "peak_count": len(peaks)
    }

print("Subject 101 Analysis:")
s101_normal = analyze_activity("101", "801-Yurume Ileri")
s101_limp = analyze_activity("101", "807-Topallama")
print(f"  Normal: {s101_normal}")
print(f"  Limping: {s101_limp}")

print("\nSubject 103 Analysis (Test):")
s103_normal = analyze_activity("103", "801-Yurume Ileri")
s103_limp = analyze_activity("103", "807-Topallama")
print(f"  Normal: {s103_normal}")
print(f"  Limping: {s103_limp}")

File: analyze_limp.js
const fs = require('fs');
const path = require('path');

// Load data from data.js using absolute path
const content = fs.readFileSync(path.join(__dirname, 'data.js'), 'utf8');
const jsonStr = content.replace('const fallData = ', '').trim().replace(/;$/, '');
const data = JSON.parse(jsonStr);

function analyzeActivity(subject, activityId) {
    const samples = data[subject][activityId]["Waist"];
    const accs = samples.map(s => Math.sqrt(s.acc_x ** 2 + s.acc_y ** 2 + s.acc_z ** 2));

    const peaks = [];
    for (let i = 1; i < accs.length - 1; i++) {
        if (accs[i] > accs[i - 1] && accs[i] > accs[i + 1] && accs[i] > 11.0) {
            peaks.push(accs[i]);
        }
    }

    const avgPeak = peaks.length > 0 ? peaks.reduce((a, b) => a + b, 0) / peaks.length : 0;

    // Calculate std dev
    const variance = peaks.length > 0
        ? peaks.reduce((a, b) => a + Math.pow(b - avgPeak, 2), 0) / peaks.length
        : 0;
    const stdPeak = Math.sqrt(variance);

    return {
        avg_peak: parseFloat(avgPeak.toFixed(2)),
        std_peak: parseFloat(stdPeak.toFixed(2)),
        peak_count: peaks.length
    };
}

console.log("Subject 101 Analysis:");
const s101Normal = analyzeActivity("101", "801-Yurume Ileri");
const s101Limp = analyzeActivity("101", "807-Topallama");
console.log("  Normal:", s101Normal);
console.log("  Limping:", s101Limp);

console.log("\nSubject 103 Analysis (Test):");
const s103Normal = analyzeActivity("103", "801-Yurume Ileri");
const s103Limp = analyzeActivity("103", "807-Topallama");
console.log("  Normal:", s103Normal);
console.log("  Limping:", s103Limp);
